### PR TITLE
[Task] [58756]  Create Video Screen

### PR DESCRIPTION
**Pull request by thaidd-1309**
## Related Tickets
- [#Task 58756: Create Ui](https://edu-redmine.sun-asterisk.vn/issues/58756)
- [#Task 58757: Handle logic](https://edu-redmine.sun-asterisk.vn/issues/58757)
- [#Task 58758: Handle data](https://edu-redmine.sun-asterisk.vn/issues/58758)
## WHAT
- Create Ui
- Loading video in cell
- Pull to refresh and load more
- Play video
- Save video 
## Evidence (Screenshot or Video)

https://user-images.githubusercontent.com/119989192/212816637-fca2dc40-09ba-4cc7-8b39-47d5b773c0fe.mp4

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*
*(Other notes)*

### CODE REVIEW

File: WallPaper_Project1/WallPaper_Project1/Data.plist [hunk 1]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+<plist version="1.0">
+<dict>
+	<key>Api_Key</key>
+	<string>563492ad6f9170000100000116158da11e414c30a7a6352f814a235a</string>
```
Comment (tunghs-1253) [inline WallPaper_Project1/WallPaper_Project1/Data.plist]: Lại đẩy API key lên rồi

File: WallPaper_Project1/WallPaper_Project1/Scenes/DetailScreen/DetailViewController.swift [hunk 2]
```diff
@@ -22,6 +25,12 @@ final class DetailViewController: UIViewController {
     @IBOutlet private weak var widthImageLabel: UILabel!
     private let apiCaller = APICaller.shared
     private var imageData: Data?
+    private var player: AVPlayer?
+    private let layerPlayer = AVPlayerLayer()
```
Comment (tunghs-1253) [inline WallPaper_Project1/WallPaper_Project1/Scenes/DetailScreen/DetailViewController.swift]:
```suggestion
    private let playerLayer = AVPlayerLayer()
```

File: WallPaper_Project1/WallPaper_Project1/Scenes/DetailScreen/DetailViewController.swift [hunk 3]
```diff
@@ -33,24 +42,59 @@ final class DetailViewController: UIViewController {
...
-        if let imageData = imageData {
-            if let imageDownload = UIImage(data: imageData) {
-                DispatchQueue.global().async {
-                    UIImageWriteToSavedPhotosAlbum(imageDownload, nil, nil, nil)
+        present(waitingLoadingViewController, animated: true)
+        if isVideo {
+            guard let url = URL(string: videoUrl) else {
+                return
+            }
+            DispatchQueue.global().async {
```
Comment (tunghs-1253) [inline WallPaper_Project1/WallPaper_Project1/Scenes/DetailScreen/DetailViewController.swift]: retain cycle

File: WallPaper_Project1/WallPaper_Project1/Scenes/VideoScreen/VideoScreenViewController.swift [hunk 4]
```diff
@@ -6,13 +6,214 @@
...
+    private func configRefesh() {
+        videoCollectionView.refreshControl = refreshControl
+        refreshControl.tintColor = .white
+        refreshControl.addTarget(self, action: #selector(refreshData(_:)), for: .valueChanged)
 
-        // Do any additional setup after loading the view.
     }
 
+    @objc private func refreshData(_ sender: Any) {
+        DispatchQueue.main.async {
```
Comment (tunghs-1253) [inline WallPaper_Project1/WallPaper_Project1/Scenes/VideoScreen/VideoScreenViewController.swift]: retain cycle

File: WallPaper_Project1/WallPaper_Project1/Widgets/LoadingReuseView/LoadCollectionReusableView 2.swift [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+//  WallPaper_Project1
+//
+//  Created by DuyThai on 12/01/2023.
+//
+
+import UIKit
+
+final class LoadCollectionReusableView: UICollectionReusableView, ReuseCell {
+
+    @IBOutlet  weak var loadViewIndicator: UIActivityIndicatorView!
```
Comment (tunghs-1253) [inline WallPaper_Project1/WallPaper_Project1/Widgets/LoadingReuseView/LoadCollectionReusableView 2.swift]: convention

File: WallPaper_Project1/WallPaper_Project1/Widgets/LoadingReuseView/LoadCollectionReusableView 2.swift [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    }
+
+    func hide() {
+        DispatchQueue.main.async {
+            self.frame = CGRect.null
+        }
+
+    }
+    func show(frame: CGRect) {
+        DispatchQueue.main.async {
```
Comment (tunghs-1253) [inline WallPaper_Project1/WallPaper_Project1/Widgets/LoadingReuseView/LoadCollectionReusableView 2.swift]: retain cycle

**Review by tunghs-1253 — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by thuannv-2723 — APPROVED** [review]

**Pull request merged**